propulsion-hotel/Domain/GroupCheckout.fs

```diff
@@ -0,0 +1,115 @@
+module Domain.GroupCheckout
+
+let [<Literal>] Category = "GroupCheckout"
+let streamId = Equinox.StreamId.gen GroupCheckoutId.toString
+let [<return: Struct>] (|StreamName|_|) = function
+    | FsCodec.StreamName.CategoryAndId (Category, GroupCheckoutId.Parse id) -> ValueSome id
+    | _ -> ValueNone
+
+module Events =
+
+    type CheckoutResidual =     { stay :  GuestStayId; residual : decimal }
+    type Event =
+        | StaysAdded of         {| at : DateTimeOffset; stays : GuestStayId[] |}
+        | StaysMerged of        {| residuals : CheckoutResidual[] |}
```

Comment:
I think this has been discussed on the discord a bit, but the terminology of "merging stays" doesn't really click with me. From my layman perspective (I don't know hotels), I had expected something along the lines of "CheckoutInitiated" as the entrypoint into the process with semantics of "I'm initiating a checkout for these twenty stays." You could still keep the functionality to add stays into the checkout in case of mistakes etc.

Reply:
A Stay can be checked out in the normal way directly by the person staying.
A Group Checkout is someone yanking a set of stays and merging them into a pending grouping.
There is a Clerk associated with the Group Checkout.
I envisage them potentially adding rooms/stays to the group in batches. In some cases, stays in the group checkout may not have been booked at the same time/in the same way, so there may not be a conclusive way to grab them all in one go (i.e. if there was a GroupBooking, the design might be different).
My naming here was based on the emphasizing "this aggregate represents a todo list of things that need to be resolved before we declare the flow complete". In that context, I thought `StaysAdded` was 'obvious'. I will a) rename it b) comment it to see if I can convey that better. Not sure ahving two aliases for the event to cover the "we are doing a group checkout with these 10 stays now" vs "oh we forgot one, add that too" is useful given that in both cases it's sort of a technical event representing "we're pulling these stays into the group now"